### PR TITLE
Introduce building parts

### PR DESCRIPTION
**Pull request by jwass**
Introduce the building part schema. Parts and main footprints share shape information.

### CODE REVIEW

**Comment by jwass** [comment]
Still todo: 
- More/better examples
- Is the building reference from part to building a string or something else?
- Should the `hasParts` boolean be on the footprint if it can be determined by inspecting the data?

**Comment by jenningsanderson** [comment]
View the change in the documentation: https://dfhx9f55j8eg5.cloudfront.net/pr/51/reference/buildings/building

**Comment by skmoore** [comment]
@jwass Looking at the PR, I can't quite tell where the building ID that the part belongs to should go. Is it just an attribute underneath "properties" like this?

"properties": {
		"building": "1234",
		"extFoo": "I am a customer user property.",
		"extBar": "Me too!",
		"theme": "buildings",
		"type": "part",
		"version": 1,
		"level": 1,
		"updateTime": "2023-06-06T10:30:00-08:00",
		"numFloors": 8,
		"sources": [{
			"property": "",
			"dataset": "OpenStreetMap"
		}]
	}

**Comment by skmoore** [comment]
@jwass Is the PR updated to show where the building ID value should go? Not sure if I missed it somewhere.

**Comment by jwass** [comment]
> @jwass Is the PR updated to show where the building ID value should go? Not sure if I missed it somewhere.

@skmoore Yes I just made the change in 051d72f0df0fea672fd2dc3ec1b0edfda08407a7. Good catch - I made it so the building id is now a required field and the example now contains the `building` field

**Comment by skmoore** [comment]
@jwass How should I submit the sample data I put together?

**Comment by jwass** [comment]
> @jwass How should I submit the sample data I put together?

@skmoore Maybe make a new directory for each one under the examples/buildings directory in the repo?

File: examples/buildings/osm/outline.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+				"value": "Valentina by Alta",
+				"language": "local"
+			}]
+		},
+		"numFloors": 8,
+		"class": "commercial",
+		"sources": [{
+			"property": "",
+			"dataset": "OpenStreetMap"
+		}]
```
Comment (jwass) [inline examples/buildings/osm/outline.json:68]: Add hasParts set to True

**Comment by jwass** [comment]
Outstanding issue: https://github.com/OvertureMaps/tf-buildings/issues/24 before merging

**Comment by LukaszMichalski-TomTom** [comment]
Should we include `roof:material` ([link](https://wiki.openstreetmap.org/wiki/Key:roof:material))  and `building:material`([link](https://wiki.openstreetmap.org/wiki/Key:building:material)) from [OSM Simple 3D Buildings](https://wiki.openstreetmap.org/wiki/Simple_3D_Buildings)?

**Comment by jwass** [comment]
> Should we include roof:material ([link](https://wiki.openstreetmap.org/wiki/Key:roof:material)) and building:material([link](https://wiki.openstreetmap.org/wiki/Key:building:material)) from [OSM Simple 3D Buildings](https://wiki.openstreetmap.org/wiki/Simple_3D_Buildings)?

@LukaszMichalski-TomTom I think we agreed at the meeting this is a good idea. Want to take a stab at adding it here?

**Comment by LukaszMichalski-TomTom** [comment]
@jwass I don't have permission to commit to this repo. I prepared the change, but I cannot push it.

**Comment by jwass** [comment]
> @jwass I don't have permission to commit to this repo. I prepared the change, but I cannot push it.

Can you make a pull request to this branch? Or can't even do that?

**Comment by jwass** [comment]
> @jwass I don't have permission to commit to this repo. I prepared the change, but I cannot push it.

Can you try again? You should have permissions now

File: schema/buildings/defs.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        - flat
+        - gable
+        - gambrel
+        - half-hip
+        - hip
+        - mansard
+        - onion
+        - pyramid
+        - round
+        - spherical
```
Comment (LukaszMichalski-TomTom) [inline schema/buildings/defs.yaml:37]: What is the difference between `spherical` and `dome`? In osm there are only 25 instances of `roof:shape=spherical` and I think they are incorrectly tagged and can be replaced with `dome`.  Example: [way:945021143](https://www.openstreetmap.org/way/945021143)

**Review by varapmsft — APPROVED** [review]

**Review by mojodna — APPROVED** [review]

File: schema/buildings/defs.yaml (hunk 2 above)
Comment (jwass) [reply schema/buildings/defs.yaml:75]: As discussed in the meeting, dome is something that is half a sphere. But sphere is needed to model something like Epcot or the new Sphere in Vegas. So we decided to keep it.

**Review by jenningsanderson — APPROVED** [review]

**Review by LukaszMichalski-TomTom — APPROVED** [review]

**Pull request merged**